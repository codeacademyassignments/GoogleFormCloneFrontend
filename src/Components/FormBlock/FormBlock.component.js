import React, { Component } from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import style from './FormBlock.style';

export default class FormBlock extends Component {
   convertUTCtoIST = utcTimeString => (new Date(utcTimeString)).toString();

  separateDateAndTime= (timeStamp) => {
    const separateDate = timeStamp.split('T');
    const separateTime = separateDate[1].split('.')[0];
    // const indianStandardTime = this.convertUTCtoIST(`${separateDate[0]} ${separateTime} UTC`).split(' ');
    // Alert.alert(`${separateDate[0]} ${separateTime} UTC`);
    return `${separateDate[0]} ${separateTime}`;
  }

  render() {
    const { formBlock: { formName, createdAt } } = this.props;
    // Alert.alert(this.separateDateAndTime(createdAt));
    return (
      <View style={style.formBlock}>
        <View style={style.innerFormBlock}>
          <Text style={style.formName}>{formName}</Text>
          <View>
            <View style={style.horizontalLine} />
            <Text style={style.creationDate}>
Created On:
              {this.separateDateAndTime(createdAt)}
            </Text>
          </View>
        </View>
      </View>
    );
  }
}

FormBlock.propTypes = {
  formBlock: PropTypes.shape({
    formName: PropTypes.string.isRequired,
    createdAt: PropTypes.string.isRequired,
  }).isRequired,
};
