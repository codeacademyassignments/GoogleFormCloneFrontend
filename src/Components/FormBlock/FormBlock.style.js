import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  formName: {
    fontSize: 22,
    fontWeight: '400',
    alignSelf: 'flex-start',
    color: 'black',
    margin: 10,
  },
  horizontalLine: {
    borderBottomColor: 'black',
    borderBottomWidth: 1,
  },
  creationDate: {
    fontSize: 18,
    alignSelf: 'flex-end',
    margin: 10,
    color: 'black',
  },
  formBlock: {
    height: 150,
    width: 300,
    backgroundColor: 'white',
    margin: 30,
  },
  innerFormBlock: {
    flex: 1,
    justifyContent: 'space-between',
  },
});
