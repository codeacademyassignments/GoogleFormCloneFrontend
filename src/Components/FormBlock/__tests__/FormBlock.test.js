import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import FormBlock from '../FormBlock.component';

describe('FormBlock', () => {
  it('should match with old snapshot', () => {
    const form = {
      formName: 'Code Academy Feedback 2019',
      createdAt: '2019-02-27T06:32:52.498Z',
    };
    const tree = renderer.create(<FormBlock formBlock={form} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('separateDateAndTime', () => {
  it('should return the passed date and time separated by space', () => {
    const form = {
      formName: 'Code Academy Feedback 2019',
      createdAt: '2019-02-27T06:32:52.498Z',
    };
    const wrapper = shallow(<FormBlock formBlock={form} />);
    expect(wrapper.instance().separateDateAndTime('2019-02-27T06:32:52.498Z')).toEqual('2019-02-27 06:32:52');
  });
});

// describe('convertUTCtoIST', () => {
//   it('should convert the utc time to ist', () => {
//     const form = {
//       formName: 'Code Academy Feedback 2019',
//       createdAt: '2019-02-27T06:32:52.498Z',
//     };
//     const wrapper = shallow(<FormBlock formBlock={form} />);
//     expect(wrapper.instance().convertUTCtoIST('2019-02-27 06:32:52 UTC')).toEqual('Wed Feb 27 2019 12:02:52 GMT+0530 (India Standard Time)');
//   });
// });
