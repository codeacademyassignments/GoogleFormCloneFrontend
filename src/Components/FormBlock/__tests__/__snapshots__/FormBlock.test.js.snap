// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBlock should match with old snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": 150,
      "margin": 30,
      "width": 300,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "black",
          "fontSize": 22,
          "fontWeight": "400",
          "margin": 10,
        }
      }
    >
      Code Academy Feedback 2019
    </Text>
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "black",
            "fontSize": 18,
            "margin": 10,
          }
        }
      >
        Created On:
        2019-02-27 06:32:52
      </Text>
    </View>
  </View>
</View>
`;
